python/src/deltachat/provider.py

```diff
@@ -0,0 +1,75 @@
+"""Provider info class."""
+
+from .capi import ffi, lib
+from .cutil import as_dc_charpointer, from_dc_charpointer
+
+
+class ProviderNotFoundError(Exception):
+    """The provider information was not found."""
+
+
+class Provider(object):
+    """Provider information.
+
+    Do not construct this directly, instead us :meth:`from_email` or
+    :meth:`from_domain`.
+    """
+
+    def __init__(self, provider):
```

Comment:
@hpk42 This API is a little odd to me.  In a way it'd be more natural if the `__init__()` had the signature of `from_domain()`.  Which could be easily done.  Hmm, even simpler than I thought when I was writing this so... let me do that.